src/libs/actions/Report.js

```diff
@@ -1484,178 +1485,136 @@ function clearIOUError(reportID) {
     Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {errorFields: {iou: null}});
 }
 
-/**
- * Internal function to help with updating the onyx state of a message of a report action.
- * @param {Object} originalReportAction
- * @param {Object} message
- * @param {String} reportID
- * @return {Object[]}
- */
-function getOptimisticDataForReportActionUpdate(originalReportAction, message, reportID) {
-    const reportActionID = originalReportAction.reportActionID;
-
-    return [
-        {
-            onyxMethod: Onyx.METHOD.MERGE,
-            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
-            value: {
-                [reportActionID]: {
-                    message: [message],
-                },
-            },
-        },
-    ];
-}
-
 /**
  * Returns true if the accountID has reacted to the report action (with the given skin tone).
- * @param {Number} accountID
- * @param {Array<Object | Number>} users
+ * Uses the NEW FORMAT for "emojiReactions"
+ * @param {String} accountID
+ * @param {Array<Object | String | number>} users
  * @param {Number} [skinTone]
  * @returns {boolean}
  */
-function hasAccountIDReacted(accountID, users, skinTone) {
-    return (
-        _.find(users, (user) => {
-            let userAccountID;
-            if (typeof user === 'object') {
-                userAccountID = user.accountID;
-            } else {
-                userAccountID = user;
-            }
-
-            return userAccountID === accountID && (skinTone == null ? true : user.skinTone === skinTone);
-        }) !== undefined
-    );
+function hasAccountIDEmojiReacted(accountID, users, skinTone) {
+    if (!skinTone) {
```

Comment:
skinTone can be `0`, right? So this condition will evaluate to true for that skinTone?

Reply:
It can, nice catch!

Reply:
Updating it to `_.isUndefined`